docs/experiments.md

```diff
@@ -192,3 +192,41 @@ This will remove quotes (`'`, `"`, and `` ` ``) from both ends of the
 metavariable content. So `"2147483648"` will be detected but `"2147483646"`
 will not. This is useful when you expect strings to contain integer or float
 data.
+
+# The Generic language
+
+Semgrep’s **generic** language matches patterns in languages that Semgrep doesn’t yet support. You can use the generic language mainline languages that don’t yet have a parser, configuration files, and other structured data such as HTML or XML.
+
+```yaml
+rules:
+- id: terraform-all-origins-allowed
+  patterns:
+  - pattern-inside: cors_rule { ... }
+  - pattern: allowed_origins = ["*"]
+  languages:
+  - generic
+  severity: WARNING
+  message: CORS rule on bucket permits any origin
+```
+
+```
+resource "aws_s3_bucket" "b" {
+  bucket = "s3-website-test-open.hashicorp.com"
+  acl    = "private"
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["PUT", "POST"]
+    allowed_origins = ["*"]  # <--- Matches here
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+  }
+}
+```
+
+
+In the code, the generic language implementation is called **spacegrep** because it tokenizes based on whitespace (and because it sounds cool 😎).
```

Comment:
"tokenizes based on whitespace" may be confusing for those who try to interpret this literally (since for example `::` is split into two `:` tokens even though there's no blank between them). I think the original desire was to have _a tool like grep but tolerant of whitespace between words_. Also, it does use indentation to identify blocks of code, but I don't think we should mention this here.